optaplanner-core/src/main/java/org/optaplanner/core/impl/constructionheuristic/DefaultConstructionHeuristicPhaseFactory.java

```diff
@@ -113,6 +108,78 @@ public DefaultConstructionHeuristicPhaseFactory(ConstructionHeuristicPhaseConfig
         return builder.build();
     }
 
+    private Optional<EntityPlacerConfig> getValidEntityPlacerConfig() {
+        return Optional.ofNullable(phaseConfig.getEntityPlacerConfig()).filter(entityPlacerConfig -> {
+            if (phaseConfig.getConstructionHeuristicType() != null) {
+                throw new IllegalArgumentException(
+                        "The constructionHeuristicType (" + phaseConfig.getConstructionHeuristicType()
+                                + ") must not be configured if the entityPlacerConfig (" + entityPlacerConfig
+                                + ") is explicitly configured.");
+            }
+            if (phaseConfig.getMoveSelectorConfigList() != null) {
+                throw new IllegalArgumentException("The moveSelectorConfigList (" + phaseConfig.getMoveSelectorConfigList()
+                        + ") cannot be configured if the entityPlacerConfig (" + entityPlacerConfig
+                        + ") is explicitly configured.");
+            }
+            return true;
+        });
+    }
+
+    private Optional<ListVariableDescriptor<?>> getValidListVariableDescriptor(HeuristicConfigPolicy<?> solverConfigPolicy) {
+        List<? extends GenuineVariableDescriptor<?>> listVariableDescriptors =
+                solverConfigPolicy.getSolutionDescriptor().findListVariableDescriptors();
+        if (listVariableDescriptors.isEmpty()) {
+            return Optional.empty();
+        }
+        if (listVariableDescriptors.size() > 1) {
+            throw new IllegalArgumentException("Construction Heuristic phase does not support multiple list variables ("
+                    + listVariableDescriptors + ").");
+        }
+
+        failIfConfigured(phaseConfig.getConstructionHeuristicType(), "constructionHeuristicType");
+        failIfConfigured(phaseConfig.getEntityPlacerConfig(), "entityPlacerConfig");
+        failIfConfigured(phaseConfig.getMoveSelectorConfigList(), "moveSelectorConfigList");
+
+        return Optional.of(((ListVariableDescriptor<?>) listVariableDescriptors.get(0)));
+    }
+
+    private static void failIfConfigured(Object configValue, String configName) {
+        if (configValue != null) {
+            throw new IllegalArgumentException("Construction Heuristic phase with a list variable does not support "
+                    + configName + " configuration. Remove the " + configName + " (" + configValue + ") from the config.");
+        }
```

Comment:
Maybe use https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-java.lang.String-.

Reply:
It seems you missed the fact that I actually require `configValue` to be null. Am I right?